test/test_torch.py

```diff
@@ -16000,6 +16010,37 @@ def test_eig(self, device, dtype):
             'A should not contain infs or NaNs',
             lambda: torch.eig(np.nan * torch.ones((2, 2))))
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.double)
+    def test_eig_out(self, device, dtype):
```

Comment:
This should also test that a warning is thrown when a non-empty out argument with the incorrect size throws a warning, and that an empty out argument is resized without a warning

Reply:
Currently, the code does not do that: it simply resizes the output tensors (as it did in the original TH code).
I seem to understand that there is a plan to change the logic to forbid passing incorrectly-sized out tensors everywhere, but I expect that once this is implemented there will be a global helper function which you can just call on each out tensor and does the resizing/warning/error/whatever.
Isn't it better to wait to have this common logic instead of writing a custom version on each operator?